core/templates/dev/head/pages/exploration_player/PlaythroughService.js

```diff
@@ -214,15 +225,22 @@ oppia.factory('PlaythroughService', [
             !this.isExplorationWhitelisted(playthrough.expId)) {
           return;
         }
-        playthrough.actions.push(LearnerActionObjectFactory.createNew(
+        var expStartLearnerAction = LearnerActionObjectFactory.createNew(
           ACTION_TYPE_EXPLORATION_START,
           {
             state_name: {
               value: initStateName
             }
           },
-          CURRENT_ACTION_SCHEMA_VERSION
-        ));
+          CURRENT_ACTION_SCHEMA_VERSION);
+        // Sometimes, the answer submit action gets recorded before the start.
+        // This happens when an answer is submitted very soon into an
+        // exploration.
+        if (playthrough.actions.length > 0) {
+          playthrough.actions.unshift(expStartLearnerAction);
+        } else {
+          playthrough.actions.push(expStartLearnerAction);
```

Comment:
I don't understand why this check is necessary; pushing and unshifting do the same thing to empty arrays. Are you sure you need this?

Reply:
Actually, true. I didn't use unshift initially, so I had to use the conditional, but unshift seems to do the same thing without the conditionals. Done